src/Dialect/ONNX/Transforms/InstrumentONNXSignaturePass.cpp

```diff
@@ -72,15 +72,34 @@ class InstrumentONNXSignaturePass
   void runOnOperation() override {
     onnx_mlir::EnableByRegexOption traceSpecificOpPattern(
         /*emptyIsNone*/ false);
+    // If the signaturePattern is started with "onnx_node_name: ",
+    // the node will be selected by the string attribute of onnx_node_name.
+    std::string header = "onnx_node_name:";
+    std::cout << signaturePattern << "\n";
+    bool useNodeName = false;
+    if (signaturePattern.rfind(header, 0) == 0) {
```

Comment:
Can you explain the behavior here? If there is one match of "onnx_node_name", then all we use the node name for all ops?

Reply:
When the option starts with `onnx_node_name:`, we use attribute("onnx_node_name"), instead of op->getName(), to check all the op.
I am not sure what you mean by node name. I modified CompilerOption.cpp for this.

Reply:
I believe the code is not doing what you think it is doing.

if the pattern passed is "onnx.Add, onnx_node_name:one_specific_op", then the `rfind` will return true and thus `useNodeName` is set to true. Then for every op in the walk, we will fetch the data from the `onnx_node_name` for alll.

I think what you need to do is:
1. use the old code to match with an op. if there is a match, then do the old print.
2. else search the onnx_node_name attribute, if non null, prefix the attribute with "onnx_node_name:", search. If hit, then do the new print including the data.

Reply:
> Then for every op in the walk, we will fetch the data from the onnx_node_name for all.

Yes, that's what I want. We could separate the pattern for onnx_node_name and getName. I did not do that.